examples/run_Hahn_echo.py

```diff
@@ -31,127 +41,100 @@
+)
+echo_data.setaxis("ph1", ph1_cyc / 4)
+echo_data.reorder(["ph1", "nScans", "t2"])
+echo_data.set_prop("postproc_type", "spincore_SE_v1")
+echo_data.set_prop("coherence_pathway", {"ph1": +1})
 echo_data.set_prop("acq_params", config_dict.asdict())
-target_directory = getDATADIR(exp_type="ODNP_NMR_comp/Echoes")
+nodename = config_dict["type"] + "_" + str(config_dict["echo_counter"])
+echo_data.name(nodename)
 filename_out = filename + ".h5"
-nodename = echo_data.name()
 if os.path.exists(f"{filename_out}"):
     print("this file already exists so we will add a node to it!")
     with h5py.File(
         os.path.normpath(os.path.join(target_directory, f"{filename_out}"))
     ) as fp:
-        if nodename in fp.keys():
-            print("this nodename already exists, so I will call it temp_echo")
-            echo_data.name("temp_echo")
-            nodename = "temp_echo"
-    echo_data.hdf5_write(f"{filename_out}", directory=target_directory)
-else:
-    try:
-        echo_data.hdf5_write(f"{filename_out}", directory=target_directory)
-    except:
-        print(
-            f"I had problems writing to the correct file {filename}.h5, so I'm going to try to save your file to temp_echo.h5 in the current directory"
-        )
-        if os.path.exists("temp_echo.h5"):
-            print("there is a temp_echo.h5 already! -- I'm removing it")
-            os.remove("temp_echo.h5")
-            echo_data.hdf5_write("temp_echo.h5")
-            print(
-                "if I got this far, that probably worked -- be sure to move/rename temp_echo.h5 to the correct name!!"
-            )
+        tempcounter = 1
+        orig_nodename = nodename
+        while nodename in fp.keys():
+            nodename = "%s_temp_%d" % (orig_nodename, tempcounter)
+            data.name(nodename)
+            tempcounter += 1
+echo_data.hdf5_write(f"{filename_out}", directory=target_directory)
 print("\n*** FILE SAVED IN TARGET DIRECTORY ***\n")
-print(("Name of saved data", echo_data.name()))
-config_dict.write()
 print(
-    "Your *current* γ_eff (MHz/G) should be ",
-    config_dict["gamma_eff_MHz_G"],
-    " - (Δν*1e-6/",
-    Field,
-    "), where Δν is your resonance offset",
+    "saved data to (node, file, exp_type):",
+    echo_data.name(),
+    filename_out,
+    my_exp_type,
 )
-print(
-    "Now run Hahn_echo_proc.py to see your spectra and offset, that script will correct your gamma effective for you!"
-)
-fl.show()
+config_dict.write()
+# }}}
+echo_data.ft("t2", shift=True)
+with figlist_var() as fl:
+    fl.next("image - ft")
+    fl.image(echo_data)
+    fl.next("image - ft, coherence")
+    echo_data.ft("ph1")
+    fl.image(echo_data)
+    fl.next("data plot")
+    if "nScans" in echo_data.dimlabels:
+        data_slice = echo_data["ph1", 1].mean("nScans")
+    else:
+        data_slice = echo_data["ph1", 1]
+    fl.plot(data_slice, alpha=0.5)
+    fl.plot(data_slice.imag, alpha=0.5)
+    fl.plot(abs(data_slice), color="k", alpha=0.5)
```

Comment:
as I commented earlier I just moved the images/plotting to after I save

Reply:
WK and AG agree it's much nicer to see the plots directly after the hahn echo as it is typically used as a diagnostic (finding signal that is way off and SW >200), running tests, etc. and while the proc_raw is useful, it's much faster/easier to just see the FTed signal directly after the ppg - so we vote to keep the plotting but move to after saving to avoid copies

Reply:
yes, but I'm also making a point, which you should not just ignore wholesale.
The obvious solution is to execute proc_raw as a shell command.
***This is not clean code***

You can use the following to do this:
```
subprocess.call(' '.join(['python ',
                          os.path.join(
                              os.path.split(
                                  os.path.split(
                                      pyspecProcScripts.__file__)[0])[0],
                                  'examples',
                                  'proc_raw.py'),
                          # add arguments here
                          ], shell=True))
```

Reply:
when messing/testing in ipython I was getting "name 'pyspecProcScripts' is not defined" so through testing and debugging arrived at using the cwd kwarg which still needs to be tested on the instrument but appears to work on my laptop